tests/std/tests/Dev10_555491_complex_linker_errors/test.cpp

```diff
@@ -47,6 +47,87 @@ bool nearly_equal_partwise(const complex<T> actual, const complex<T> expected) {
     return nearly_equal(actual.real(), expected.real()) && nearly_equal(actual.imag(), expected.imag());
 }
 
+// Also test GH-2728 <complex>: Implementation divergence for division by zero
+void test_gh_2728() {
+    const complex<double> zero{0, 0};
+    {
+        const complex<double> test_one_one = complex<double>{1, 1} / zero;
+        assert(isinf(test_one_one.real()) && test_one_one.real() > 0);
+        assert(isinf(test_one_one.imag()) && test_one_one.imag() > 0);
+    }
+    {
+        const complex<double> test_one_zero = complex<double>{1, 0} / zero;
+        assert(isinf(test_one_zero.real()) && test_one_zero.real() > 0);
+        assert(isnan(test_one_zero.imag()));
+    }
+    {
+        const complex<double> test_one_minus_one = complex<double>{1, -1} / zero;
+        assert(isinf(test_one_minus_one.real()) && test_one_minus_one.real() > 0);
+        assert(isinf(test_one_minus_one.imag()) && test_one_minus_one.imag() < 0);
+    }
+    {
+        const complex<double> test_zero_one = complex<double>{0, 1} / zero;
+        assert(isnan(test_zero_one.real()));
+        assert(isinf(test_zero_one.imag()) && test_zero_one.imag() > 0);
+    }
+    {
+        const complex<double> test_zero_zero = complex<double>{0, 0} / zero;
+        assert(isnan(test_zero_zero.real()));
+        assert(isnan(test_zero_zero.imag()));
+    }
+    {
+        const complex<double> test_zero_minus_one = complex<double>{0, -1} / zero;
+        assert(isnan(test_zero_minus_one.real()));
+        assert(isinf(test_zero_minus_one.imag()) && test_zero_minus_one.imag() < 0);
+    }
+    {
+        const complex<double> test_minus_one_one = complex<double>{-1, 1} / zero;
+        assert(isinf(test_minus_one_one.real()) && test_minus_one_one.real() < 0);
+        assert(isinf(test_minus_one_one.imag()) && test_minus_one_one.imag() > 0);
+    }
+    {
+        const complex<double> test_minus_one_zero = complex<double>{-1, 0} / zero;
+        assert(isinf(test_minus_one_zero.real()) && test_minus_one_zero.real() < 0);
+        assert(isnan(test_minus_one_zero.imag()));
+    }
+    {
+        const complex<double> test_minus_one_minus_one = complex<double>{-1, -1} / zero;
+        assert(isinf(test_minus_one_minus_one.real()) && test_minus_one_minus_one.real() < 0);
+        assert(isinf(test_minus_one_minus_one.imag()) && test_minus_one_minus_one.imag() < 0);
+    }
+    {
+        const complex<double> test_inf_inf = complex<double>{INFINITY, INFINITY} / zero;

```

Comment:
We generally avoid the `INFINITY` and `NAN` macros in test code (the CRT defines them in a somewhat unusual way and this has changed over time; `numeric_limits` uses builtins with predictable behavior); below this test uses `constexpr double inf = numeric_limits<double>::infinity();` and we could do the same here.